src/v/storage/segment_deduplication_utils.h

```diff
@@ -60,4 +60,11 @@ ss::future<index_state> deduplicate_segment(
   ss::sharded<features::feature_table>&,
   bool inject_reader_failure = false);
 
+ss::future<> linear_scan_of_segment_for_map(
```

Comment:
nit: please add comments about what this is doing

Reply:
Added comments, and also renamed this function.